### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by moise-mulungu**
On this milestone, I:
- Installed devise 
- Configured devise in the Gem file
- The devise should :
  * Register a new user
  * Store  hashed password in the database
  * Get User logs in with a combination of email and password
  * Ask for confirmation of email
  * Reset password
- Updated the README file

Screenshot from 2022-07-20 23-26-28
Screenshot from 2022-07-20 23-29-26
Screenshot from 2022-07-20 23-30-34
Screenshot from 2022-07-20 23-31-28

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (prolajumokeoni) [inline app/models/user.rb]: - [ ] Kindly enable the application to ask for confirmation of email by enabling `:confirmable` in the user model 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @moise-mulungu, 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- The views in devise for the registration and login should be modified to match the wireframes and you can personalize the styling. You can take a look at the wireframes in the [sneak peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md) of the project

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
```
Comment (Davidosky007) [inline app/views/devise/registrations/edit.html.erb:10]: * Due to the changes made in respect of the previous reviewer's request your code now has a bug, as when i try signing up i get an error, Kindly fix this. below  is a screenshot of the said error i get ⬇️ 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @moise-mulungu @kaitcham 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* No linter errors 👍 
* Well written readme file 👍 
* Good looking PR description 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by moise-mulungu** [comment]
## Hello Davidosky007 ,
When I tried to add `: confirmable` in the user model, I did not check the app because I thought it might work as before, but now I realize I need to change quite a lot in my code. Please can you spare some minutes to assist me because honestly, I cannot fix that? 
Thanks for your understanding

**Review by aimemalaika — APPROVED** [review]
Hi @moise-mulungu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**